lib/api/BackbeatAPI.js

```diff
@@ -157,13 +162,29 @@ class BackbeatAPI {
             bbRequest.setMatchedRoute(promRoute);
             return null;
         }
+        console.log('routes', routes);
+
+        // If giving the bucket name, the user must also provide an object key
+        // and version ID.
+        const hasGranularity = rDetails.bucketName ?
+            rDetails.objectKey && rDetails.versionId : true;
+        if (!hasGranularity) {
+            return errors.RouteNotFound.customizeDescription(
+                'must provide object key and version ID in route: ' +
+                `${bbRequest.getRoute()}`);
+        }
 
         // All routes at this point have extensions property in use
         // Match http verb and match extension name
-        let filteredRoutes = routes.filter(r => (
-            bbRequest.getHTTPMethod() === r.httpMethod &&
-            r.extensions[rDetails.extension]
-        ));
+        let filteredRoutes = routes.filter(r => {
+            const isCorrectMethod = bbRequest.getHTTPMethod() === r.httpMethod;
```

Comment:
minor naming issue: "correct" may not be the appropriate wording, since we are trying to find a route that "matches" what we provide, it's not really "incorrect" not to match a specific route. Maybe `isMatchingXxx` instead would work better.